shell/common/animator.cc

```diff
@@ -106,34 +99,29 @@ void Animator::BeginFrame(
   FML_DCHECK(producer_continuation_);
   const fml::TimePoint frame_target_time =
       frame_timings_recorder_->GetVsyncTargetTime();
-  dart_frame_deadline_ = FxlToDartOrEarlier(frame_target_time);
+  dart_frame_deadline_ = frame_target_time.ToEpochDelta();
   uint64_t frame_number = frame_timings_recorder_->GetFrameNumber();
   delegate_.OnAnimatorBeginFrame(frame_target_time, frame_number);
 
   if (!frame_scheduled_ && has_rendered_) {
-    // Under certain workloads (such as our parent view resizing us, which is
-    // communicated to us by repeat viewport metrics events), we won't
-    // actually have a frame scheduled yet, despite the fact that we *will* be
-    // producing a frame next vsync (it will be scheduled once we receive the
-    // viewport event).  Because of this, we hold off on calling
-    // |OnAnimatorNotifyIdle| for a little bit, as that could cause garbage
-    // collection to trigger at a highly undesirable time.
+    // Wait a tad more than 3 60hz frames before reporting a big idle period.
```

Comment:
Do we have any logic in place that prevents having multiple outstanding NotifyIdle calls scheduled? Said differently, is there logic to ensure that there is a maximum of one call at a time scheduled to run? If that property is a side-effect of something else, would it be worth making it explicit?

Reply:
The old logic said if `RequestFrame` and `BeginFrame` have not been called again for 51ms, fire a big idle notification. The new logic says if `RequestFrame` has not been called in the last 51ms and the last known vsync target time is not in the future, fire.

We could still easily end up in a situation where `NotifyIdle` is called at the end of a frame and then called again ~51ms later if there are no more frames.

The situation we are still not avoiding is that we could call `NotifyIdle` at the end of a very cheap frame only to immediately start pumping an animation/route transition in the very next frame.

Reply:
And to answer your specific question: we could schedule multiple tasks, but the guards in those tasks would prevent notify idle from being called repeatedly (i.e. more than once per frame or every 51ms when no frames are pumped).